biologging-sensor-data/src/components/graphs/line/LineGraph.tsx

```diff
@@ -73,19 +79,52 @@ export default function LineGraph({ events, sensor }: { events: Event[], sensor:
   }, [events]);
 
   useEffect(() => {
-    setOptions({
-      responsive: true,
-      plugins: {
-        legend: {
-          position: 'bottom' as const,
-        },
-        title: {
-          display: true,
-          text: sensor.toUpperCase(),
-        },
+    const fetchData = async () => {
+      try {
+        const response = await get<any>(`dataset/${events[0].datasetID}`);
```

Comment:
Can you change this to a relevant API  file? please, just check page.tsx in Detail, line 14, and see how the archietcture is defined.

So all API calls shoudl be defined in API file in api folder (in this case api/instrument/api.ts, you should create it if it is not there ), create a function and call it from here. Please, check api/dataset/api.ts, function getDataset for reference.

Reply:
@zrohacova ,  I will try to refactor this part of the code.